azkaban-common/src/main/java/azkaban/imagemgmt/version/VersionSet.java

```diff
@@ -64,11 +65,11 @@ public VersionSet(final String versionSetJsonString, final String versionSetMd5H
           .withAdditionalKeyDeserializers(new CaseInsensitiveKeyDeserializers());
       mapper.setDeserializerProvider(provider);
       this.imageToVersionMap = mapper.readValue(this.versionSetJsonString,
-          new TypeReference<TreeMap<String, String>>() {
+          new TypeReference<TreeMap<String, VersionInfo>>() {
```

Comment:
I believe this requires extending the versionSetJsonString to contain the path and state in addition to the version. 
Is that correct or does this information directly fetched from some database columns ?

Reply:
Yes versionSetJsonString is JSON representation (in string format) of VersionSet and it is stored in version_set table as JSON. Yes this also includes path and state in addition to version.